node/stateSync.go

```diff
@@ -171,10 +172,14 @@ func (s *stateSync) run() error {
 
 func (s *stateSync) startWorkerThread() {
 	var workerId uint32
-	peersNum := len(s.peers)
-	for workerId = 0; workerId < uint32(peersNum*concurrentSyncRequestPerNeighbor); workerId++ {
+	peersNum := uint32(len(s.peers) * concurrentSyncRequestPerNeighbor)
```

Comment:
This will cause L182 to use the wrong peersNum. Try something like this (you might need to fix type)
```
peersNum := len(s.peers)
workerNum := peersNum * concurrentSyncRequestPerNeighbor
if workerNum > config.Parameters.SyncStateMaxPeer {
 	workerNum = config.Parameters.SyncStateMaxPeer
 }
```

Reply:
Fixed